api/proxy/servers/arbitrum_altda/server.go

```diff
@@ -0,0 +1,101 @@
+package arbitrum_altda
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// determine which upstream example DA Server config fields are absolutely necessary.
+// the example DA Provider server provided by OCL uses:
+//   - String JWT auth
+//   - Write toggling
+//   - Server request body limits
+//
+// JWT authorization is supported by the AnyTrust and example Custom DA servers
+// and toggled in core nitro as a client config field
+// TODO: Add support for JWT authentication
+//
+// Write toggling allows a user to use the ALT DA server in "read only" mode
+// which is typical behavior for all rollup operator nodes outside of the batch poster
+// TODO: Add support for "read only" mode
+//
+// Server request body limits are used to impose a maximum on the allowed request body size
+// TODO: Understand if this is actually used/respected in DA client anywhere
+// TODO: Add env ingestion for these values
+// TODO: Determine a proper default given EigenDA's large batch uniqueness. The defaultBodyLimit
+// //    is currently set to 5 mib which is too low for integration EigenDA
+//
+// The ALT DA server implementation should be a thin wrapper over the existing
+// storage abstractions with lightweight translation from the existing critical
+// REST status code signals (i.e, "drop cert", "failover") into arbitrum specific
+// errors
+type Config struct {
+	Enable bool
+	Host   string
+	Port   int
+}
+
+type Server struct {
+	cfg  *Config
+	svr  *http.Server
+	addr string
+}
+
+// NewServer constructs the RPC server
+func NewServer(ctx context.Context, cfg *Config) (*Server, error) {
+	rpcServer := rpc.NewServer()
+	if err := rpcServer.RegisterName("daprovider", &Handlers{}); err != nil {
+		return nil, fmt.Errorf("failed to register daprovider: %w", err)
+	}
+
+	svr := &http.Server{
+		Handler: rpcServer,
+	}
+
+	return &Server{
+		cfg: cfg,
+		svr: svr,
+	}, nil
+
+}
+
+func (s *Server) Addr() string {
+	return s.addr
+}
```

Comment:
this is unused. is it really needed?

Reply:
oh I think you just forgot to use it in the log line where the server gets started